packages/config/src/check.ts

```diff
@@ -0,0 +1,58 @@
+import { getSchemaErrors, recursiveSidebarIterate } from '@/utils'
+import type { Static } from '@sinclair/typebox'
+import { Value, type ValueError } from '@sinclair/typebox/value'
+import fs from 'node:fs'
+
+import { ScalarConfigSchema, sidebarSchemaMap } from './configTypes'
+
+/** check scalar config file using the generated schema */
+export function check(file: string) {
+  try {
+    const scalarConfigFile = fs.readFileSync(file, 'utf8')
+    const scalarConfigJson = JSON.parse(scalarConfigFile)
+
+    const result = Value.Check(ScalarConfigSchema, scalarConfigJson)
+    const errors = [...Value.Errors(ScalarConfigSchema, scalarConfigJson)]
+
+    // Early return if there is already an error with the schema at the root level
+    if (!result) {
+      return {
+        valid: errors.length === 0,
+        errors: errors,
+      }
+    }
```

Comment:
Opinions on this early return? It is useful for type safety in the rest of the function.

Reply:
Great